his_checker.py

```diff
@@ -9,9 +9,37 @@
 
 import sys
 import argparse
+import os
+import re
+
+PARAMETERS_THRESHOLD = 5
+
+def process_parameters(files):
+
+    """Process the number of function parameters."""
+
+    metric_fail = 0
+    cflow = "cflow -l --depth=1"
+    print("Checking the number of function parameters")
+    for file in files.files:
+        with os.popen(f"{cflow} {file}") as pipe:
+            lines = pipe.read()
+        pattern = r'\{.*?\} (\w+)\(.*?\(([^)]*)\) at ([^:]+):(\d+)'
+        matches = re.findall(pattern, lines)
+        for function_name, parameters, file_path, line_number in matches:
+            parameters = \
+            [param.strip() for param in parameters.split(',') if param.strip()]
+            num_param = len(parameters)
+            if num_param > PARAMETERS_THRESHOLD:
+                print(f"At {file_path}({line_number}): {function_name} has "
+                      f"{num_param} function parameters. Functions must only "
+                      f"have a maximum of {PARAMETERS_THRESHOLD} parameters.")
```

Comment:
Phrase it as in https://github.com/bao-project/bao-ci/pull/74/files to maintain consistency

"(...)The maximum is X parameters."

Reply:
Done